src/nova_ws_handler.erl

```diff
@@ -40,8 +40,23 @@ init(Req, State = #{mod := Mod, controller_data := ControllerData}) ->
             ?ERROR("Websocket handler ~p returned unkown result ~p", [Mod, Error]),
             Req1 = cowboy_req:reply(500, Req),
             {ok, Req1, State}
+    end;
+init(Req, State) ->
+    %% We have something that needs securing
+    case nova_security_plugin:pre_http_request(State, #{}) of
```

Comment:
In this case could we add a pre_ws_upgrade that we could have in our plugins? Then here we could apply pre_ws_plugins in correct prio order the same that we do for http request and for ws?

I think that would be a cleaner solution if you want to have more things before upgrading to a websocket.

Reply:
We already have support for a `pre_ws_request` so this should probably be this. Or a `pre_upgrade_ws_request`. I've already started on such a solution :)

Reply:
yeah, meant pre_upgrade_ws_request.